src/aura-fed/AuraFed.sol

```diff
@@ -4,226 +4,158 @@ import "src/interfaces/IERC20.sol";
-        uint bptValue = bptSupply() * bpt.getRate() / 10**18;
-        if(harvestLP && bptValue > dolaSupply) {
+        uint bptValue = claimsSupply() * BPT.getRate() / 10**18;
+        if(flag == 1 && bptValue > debt) {
             require(msg.sender == chair, "ONLY CHAIR CAN TAKE BPT PROFIT");
-            uint dolaSurplus = bptValue - dolaSupply;
+            uint dolaSurplus = bptValue - debt;
             uint bptToWithdraw = bptNeededForDola(dolaSurplus);
             if(bptToWithdraw > dolaBptRewardPool.balanceOf(address(this))){
                 bptToWithdraw = dolaBptRewardPool.balanceOf(address(this));
             }
             require(dolaBptRewardPool.withdrawAndUnwrap(bptToWithdraw, false), "AURA WITHDRAW FAILED");
-            uint dolaProfit = _withdraw(dolaSurplus, maxLossTakeProfitBps);
+            uint dolaProfit = _removeLiquidity(dolaSurplus, maxLossTakeProfitBps);
             require(dolaProfit > 0, "NO PROFIT");
-            dola.transfer(gov, dolaProfit);
+            DOLA.transfer(gov, dolaProfit);
         }
 
         require(dolaBptRewardPool.getReward(address(this), true), "Getting reward failed");
-        bal.transfer(gov, bal.balanceOf(address(this)));
-        aura.transfer(gov, aura.balanceOf(address(this)));
+        uint balBalance = BAL.balanceOf(address(this));
+        uint auraBalance = AURA.balanceOf(address(this));
+        if(balBalance > 0) BAL.transfer(gov, balBalance);
+        if(auraBalance > 0) AURA.transfer(gov, auraBalance);
     }
 
     /**
-    @notice Burns the remaining dola supply in case the FED has been completely contracted, and still has a negative dola balance.
-    */
-    function burnRemainingDolaSupply() public {
-        dola.transferFrom(msg.sender, address(this), dolaSupply);
-        dola.burn(dolaSupply);
-        dolaSupply = 0;
+     * @notice Withdraws balancer pool tokens to governance in an emergency.
+     * @dev Will ruin accounting of the contract
+     */
+    function emergencyWithdraw() onlyGov override external {
+        require(dolaBptRewardPool.withdrawAndUnwrap(dolaBptRewardPool.balanceOf(address(this)), false), "AURA WITHDRAW FAILED");
+        BPT.transfer(gov, BPT.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Migrates claims tokens and debt to the Migrator fed contract
+     * @dev The calling migrator must increment its debt by amount returned by this call
+     * @param claimsToMigrate Amount of claims tokens to migrate
+     * @return Amount of debt to migrate
+     */
+    function migrateTo(uint claimsToMigrate) onlyMigrator override external returns(uint){
+        uint totalClaims = claimsSupply();
+        require(totalClaims >= claimsToMigrate, "NOT ENOUGH CLAIMS");
+        uint claimsToUnstake =  claimsToMigrate - BPT.balanceOf(address(this));
+        require(dolaBptRewardPool.withdrawAndUnwrap(claimsToUnstake, false), "AURA WITHDRAW FAILED");
+        uint debtToMigrate = debt * claimsToMigrate / totalClaims;
```

Comment:
is it possible that there is ever `debt > 0`, and also `totalClaims == 0`? 

If so this would revert when dividing by zero and wouldn't allow us to move that debt from one Fed to another.

Reply:
It's possible but at that point there would be no point in migrating, as the debt is purely bad debt and it has no claims to back it up. Becomes the job of the treasury to pay off the bad debt accumulated by the Fed.